testhelper/docker/resource/minio.go

```diff
@@ -0,0 +1,119 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/ory/dockertest/v3"
+
+	"github.com/rudderlabs/rudder-go-kit/httputil"
+	minioconfig "github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource/minio"
+)
+
+type MinioResource struct {
+	BucketName      string
+	AccessKeyID     string
+	AccessKeySecret string
+	Endpoint        string
+	Region          string
+	Client          *minio.Client
+}
+
+func (mr *MinioResource) ToFileManagerConfig(prefix string) map[string]any {
+	return map[string]any{
+		"bucketName":       mr.BucketName,
+		"accessKeyID":      mr.AccessKeyID,
+		"secretAccessKey":  mr.AccessKeySecret,
+		"accessKey":        mr.AccessKeySecret,
+		"enableSSE":        false,
+		"prefix":           prefix,
+		"endPoint":         mr.Endpoint,
+		"s3ForcePathStyle": true,
+		"disableSSL":       true,
+		"useSSL":           false,
+		"region":           mr.Region,
+	}
+}
+
+func SetupMinio(pool *dockertest.Pool, d cleaner, opts ...func(*minioconfig.Config)) (*MinioResource, error) {
+	const (
+		bucket          = "rudder-saas"
```

Comment:
Should we name it `testbucket` instead of RudderStack production bucket?
```suggestion
		bucket          = "testbucket"
```